### ISSUE TITLE
Plan features page takes users out of the blogger onboarding flow

### ISSUE THREAD

**Issue by roo2** [Automattic/wp-calypso#56303]
With an _Existing_ user, start from https://wordpress.com/start and select the "Personal" or "Business" plan. Proceed through checkout.

Expected:
Any user going through the /start flow should proceed to the Intent Screen of the Hero Onboarding Flow once the plan selection/checkout flow is finished.
Note: this will be a change to the "Personal" or "Business" plan flows, eliminating this screen from those flows in favor of sending the user through the Hero Flow.

Actual:
You are shown a congratulations prompt which takes you out of the onboarding flow.
<img width="974" alt="Screen Shot 2021-09-16 at 11 53 51 am" src="https://user-images.githubusercontent.com/22446385/133538059-d52f842d-2b0a-45c3-b91e-195ff452c6d6.png">

**Comment by p-jackson** [comment]
Interesting I get a Quick Start Session up sell on the Personal plan.

<img width="816" alt="Screenshot 2021-09-16 at 3 52 38 PM" src="https://user-images.githubusercontent.com/1500769/133546766-b1d231a9-21eb-4bac-8755-5bc3bb185690.png">

If you say you don't want a quick start session you're taken back to the design picker.

My understanding was that we didn't want to change any of these flows where a potential upsell happens. So for example if the user clears their cart without paying then they're taken to the `/plans` page in Calypso, which is what currenctly happens in production. That way it's an option to select another plan. The hero flow we're experimenting with now isn't to interfere with any of the current way sales are generated, am I right on that @ollierozdarz @autumnfjeld?

**Comment by ollierozdarz** [comment]
Hey @p-jackson, thanks for calling this out. 

I went through the flow just now and clicking 'No thanks' took me to My Home. But I think you're saying once in checkout for a Quick start session, then it's taking you to the plans page. 

There are a few edge cases around purchasing a plan or domain. We should go through and try to handle them on a case by case basis. @razvanpapadopol graciously documented them in the Pod Slack channel, but they are again:

——————

When the user successfully completes a purchase:

(a.) Without domain, they can land in some upsell pages or a thank you page
(b.) With a paid domain, they land on a thank you page with some links and instructions to configure their domain
(c.) With email, I don’t know yet but must be some config as well (**Note:** I'm unsure if email is still being testing in the onboarding flow so this may not be an issue?)
(d.) There was an eCommerce one, but we're not touching the 'Sell' flow for now so it won't apply.

When completing a, b, c, users land in My Home (currently).

**My thoughts on above:**

I'd love to know if users in general actually purchase Quick Start sessions at this very early stage. Is this something we're able to figure out?

If they are, we should respect it and possibly apply a fresh coat of paint and change the redirects to either go back to the /start flow if they cancel or on to the hero flow if successful. If they aren't purchasing these sessions here, it seems like the perfect candidate to put out to pasture.

——————

When the user abandons Checkout: they are redirected to the Plans page in Calypso where they can pick a different plan and they also see the Cart button with a pulsating badge showing the number of products in the cart.


**My thoughts on above:**

We had a lengthy discussion about this in Slack a few weeks back, and I feel pretty strongly that if a user abandons the cart, they should remain in the signup flow and go back to the Plans page (../start/plans). This way they stay in the flow and  still get a chance to see the hero flow without getting lost in the void of Calypso with context.

**Comment by p-jackson** [comment]
> I'd love to know if users in general actually purchase Quick Start sessions at this very early stage. Is this something we're able to figure out?

I'm sure we can, I'll look into it.

> if a user abandons the cart, they should remain in the signup flow and go back to the Plans page (../start/plans)

I agree except ... I personally think the free option on `/start/plans` is hard to find. I'm sure it's intentional, de-emphasising the free plan means it'll be more likely folks will pick a paid plan. But what I think this means is that plenty of users who make it to checkout actually wanted the free plan. Emptying the cart and landing on the dashboard is the flow we've sort of encouraged them to use. If emptying the cart takes them back to `/start/plans` then we're taking them back to the page they already couldn't figure out how to use (through no fault of their own).

**Comment by ollierozdarz** [comment]
> I agree except ... I personally think the free option on /start/plans is hard to find.

Agree totally about the Free plan. The implementation as we know it now is actually not the ideal state and not intentional. Ideally it's [this](https://d.pr/i/KFKZAq) (which is my original design), but I'm also aware that the Pricing and Packaging pod have some changes for this page incoming which should also make Free easier to find too.

I think you make a good point but it's a bit of a bandaid solution around Free currently being hidden. I'd rather fix the root cause (Free not being obvious) and continue on with our ideal flow 👯‍♀️.

**Comment by p-jackson** [comment]
> I'd love to know if users in general actually purchase Quick Start sessions at this very early stage. Is this something we're able to figure out?

pdgK6S-88-p2

> I'd rather fix the root cause (Free not being obvious) and continue on with our ideal flow 👯‍♀️.

Yes please!! 🕺

**Comment by roo2** [comment]
>Interesting I get a Quick Start Session up sell on the Personal plan.

Ahh ok maybe this issue only affects personal and business plan for _existing_ users if new ones are seeing the quick start sessions.

**Comment by autumnfjeld** [comment]
Am I understanding the conclusion here: 

* we keep all upsell flows
* if user completes upsell flow we respect screens that give them options (like the screen posted by @roo2 ) that will mean they don't get routed into the new design picker flow (or any flow we are doing in this blogger pod iteration)
   * if that screen doesn't have options (just a close or continue button like the screen posted by @p-jackson above) then we'll redirect the user to the design picker flow when they click close/continue
* if a user aborts the upsell then they either get routed back to /start/plans (or they continue on with design picker flow if that makes sense)

**Comment by ollierozdarz** [comment]
@autumnfjeld I'm not sure there's a conclusion yet. Let's give @p-jackson's post a few days to be seen by the right people, then we can deice the route. 

In a nutshell, I think this upsell can either be moved or removed to fit with out new flow. I don't think we should leave it as is (unless there's a very compelling reason), because it defeats the purpose of all the work we're doing to rethink the journey of a new WordPress.com user.

**Comment by roo2** [comment]
I think that there are two issues worth talking about here.
The first is the issue that you have been discussing about the Quick Start sessions upsell which is shown to new users. pdgK6S-88-p2

The second is the original issue which this ticket described: 

I think that it is an existing issue with the plan purchase congratulations message being shown in the start flow for existing users. I believe that the congratulations message is only intended to be shown when a user upgrades their existing plan via the /plans page. 

If the existing user uses the /start flow to create a new site, and purchases a personal or business plan, they will see the congratulations message immediately after purchase. But if they were in a flow that has steps after purchasing a plan (such as this new signup/setup-site-after-checkout flow), then this page interrupts the flow and can only be exited by going to customer home.

A third issue which came up is that these upsell messages have been redesigned, new designs have been implemented for some congratulations pages but not others pdgK6S-7f-p2#comment-284

**Comment by autumnfjeld** [comment]
@jaswrks Just verifying this is now in the Martech backlog. :). Thank you!

**Comment by roo2** [comment]
fixed by #59499
